lib/internal/Magento/Framework/Module/Registrar.php

```diff
@@ -32,6 +32,11 @@ public static function registerModule($moduleName, $path)
         self::$modulePaths[$moduleName] = $path;
     }
 
+    public static function unregisterModule($moduleName)
```

Comment:
I don't think this is a good idea. What happens if this method is called when the application has already been bootstrapped and made use of a module? I think we need to make these objects immutable (see #1562), and tests should use mocks if they need to make use of the a module registry